src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockBuilder.cs

```diff
@@ -75,6 +83,12 @@ public TagHelperBlockBuilder(TagHelperBlock original)
         }
 
         /// <summary>
+        /// Text used as a required prefix when matching HTML start and end tags in the Razor source to available 
+        /// tag helpers.
+        /// </summary>
+        public string Prefix { get; set; }
```

Comment:
Will make a `FullTagName` property, remove this and the `FullTagName` property will calculate it on the fly by looking at the descriptors. This way it makes the builder accurate depending on its descriptors and reduces the amount of work we do outside of the builder.